kolibri/plugins/coach/assets/src/views/plan/QuizSummaryPage/api.js

```diff
@@ -1,17 +1,52 @@
 import { fetchExamWithContent } from 'kolibri.utils.exams';
 import { ExamResource } from 'kolibri.resources';
+import QuizDifficulties from '../../../apiResources/quizDifficulties';
+import { getDifficultQuestions } from '../../../utils';
 
-export function fetchQuizSummaryPageData(examId) {
-  return ExamResource.fetchModel({ id: examId })
-    .then(exam => {
-      return fetchExamWithContent(exam);
-    })
-    .then(({ exam, exercises }) => {
-      return {
-        exerciseContentNodes: exercises,
-        exam,
-      };
-    });
+const fetchDifficultQuestions = async exam => {
+  if (exam.draft) {
+    return [];
+  }
+  const correctnessStats = await QuizDifficulties.fetchDetailCollection(
+    'detail',
+    exam.id,
+    undefined,
+    true,
+  );
+
+  const allQuestions = exam.question_sources.reduce((qs, section) => {
+    qs = [...qs, ...section.questions];
+
+    return qs;
```

Comment:
Non-blocking: You could  `return [...qs, ...section.questions]` here rather than assigning to the `qs` and returning it.

Reply:
Oh yes, that would be much more simpler. Thanks!